addons/customize-avatar-border/addon.json

```diff
@@ -95,9 +170,58 @@
           "fill-transparent": true
         }
       }
+    },
+    {
+      "name": "Custom shape",
+      "id": "rounding",
+      "type": "boolean",
+      "default": false
+    },
+    {
+      "name": "Shape",
+      "id": "round-style",
+      "type": "select",
+      "potentialValues": [
+        {
+          "id": "square",
+          "name": "Square"
+        },
+        {
+          "id": "round",
+          "name": "Rounded square"
+        },
+        {
+          "id": "circle",
+          "name": "Circle"
+        }
+      ],
+      "default": "round",
+      "if": {
+        "settings": {
+          "rounding": true
+        }
+      }
+    },
+    {
+      "name": "Corner radius (%)",
+      "id": "radiusSetting",
+      "type": "integer",
+      "min": 1,
+      "max": 49,
+      "default": 12,
+      "if": {
+        "settings": {
+          "rounding": true,
+          "round-style": "round"
+        }
+      }
```

Comment:
I'm trying to only show the percentage setting when both `rounding` is true and `round-style` is "round", but it shows the setting when at least one is true. Is this a bug? If not, how do I make this an AND instead of an OR?

Reply:
@WorldLanguages Do you know if this behavior is a bug or something I'm doing?

Reply:
> `if` currently always uses `OR` and not `AND` when multiple conditions are listed - see [#3267 (comment)](https://github.com/ScratchAddons/ScratchAddons/issues/3267#issuecomment-895323638). This could be changed, but in this case it's probably better to replace `rounding` and `round-style` with a single `select` setting. The options could be "Default", "Square", "Round", "Circle".

Alright, I'll do that.

I looked at the documentation and I thought it said that every condition needed to be true, unless I didn't look close enough or there was a mistake.

Reply:
Actually, this might be a bug, because `if` for userscripts does work as described in the docs.

Reply:
@mxmou Any more details on the bug?

Reply:
> @mxmou Any more details on the bug?

The following code can be used to demonstrate the issue:
<details><summary>Show code</summary>
<p>

`addon.json:`
```json
{
  "name": "Test",
  "description": "Test.",
  "tags": ["community"],
  "dynamicEnable": true,
  "dynamicDisable": true,
  "updateUserstylesOnSettingsChange": true,
  "userstyles": [
    {
      "url": "test.css",
      "matches": ["scratchWWWNoProject"],
      "if": {
        "settings": {
          "checkbox1": true,
          "checkbox2": true
        }
      }
    }
  ],
  "settings": [
    {
      "id": "checkbox1",
      "type": "boolean",
      "name": "Checkbox 1",
      "default": false
    },
    {
      "id": "checkbox2",
      "type": "boolean",
      "name": "Checkbox 2",
      "default": false
    },
    {
      "id": "checkbox3",
      "type": "boolean",
      "name": "Checkbox 3",
      "default": false,
      "if": {
        "settings": {
          "checkbox1": true,
          "checkbox2": true
        }
      }
    }
  ]
}
```

`test.css`:
```css
#navigation {
  background-color: orange !important;
}
```

</p>
</details>

The condition for the userstyle and the "Checkbox 3" setting is identical, but they behave differently:
* the userstyle is injected if both "Checkbox 1" and "Checkbox 2" are turned on,
* "Checkbox 3" appears if at least one of the other two settings is turned on.

Reply:
I opened an issue: #5692